cql3/expr/prepare_expr.cc

```diff
@@ -817,17 +817,36 @@ cast_prepare_expression(const cast& c, data_dictionary::database db, const sstri
 
 std::optional<expression>
 prepare_function_call(const expr::function_call& fc, data_dictionary::database db, const sstring& keyspace, const schema* schema_opt, lw_shared_ptr<column_specification> receiver) {
-    if (!receiver) {
-        // TODO: It is possible to infer the type of a function call if there is only one overload, or if all overloads return the same type
+    if (!schema_opt && !receiver) {
+        // In order to prepare a function_call we need to know the keyspace and column family (table name),
+        // which define the environment where this function is defined and executed.
+        // The information about keyspace is provided as an argument, but the information about column_family
+        // has to be acquired either from schema_opt or receiver.
+        // When both of them are not available it's impossible to prepare the function.
         return std::nullopt;
```

Comment:
That's only true for the token function, no?

Reply:
Right, I will change it so that the column family is required only for the `token()` function.